api/libproject/cloudsqlproxycontext.rb

```diff
@@ -0,0 +1,27 @@
+require_relative "../../libproject/utils/common"
+require_relative "credentials"
+require_relative "gcloudcontext"
+
+class CloudSqlProxyContext
+  attr_reader :env_file_path
+
+  def run(gcloud_context)
```

Comment:
I think only the `env_key` from `gcloud_context` is used -- would it make sense to only pass that in? `env_key` is `:dev` or (later) `:prod`?

I like simplicity of passing in just one thing, but it's a lot of mental burden to figure out what `CloudSqlProxyContext`'s actual dependencies are.